enclave/sgx/verifier.c

```diff
@@ -0,0 +1,349 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "verifier.h"
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/crypto/sha.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include <openenclave/internal/safemath.h>
+#include <stdlib.h>
+#include <string.h>
+#include "platform_t.h"
+#include "report.h"
+
+// Hard code Intel signed QvE Identity below
+// You can get such info from latest QvE Identity JSON file
+// e.g., Get the QvE Identity JSON file from
+// https://api.trustedservices.intel.com/sgx/certification/v2/qve/identity
+//
+
+static const uint32_t g_qve_misc_select = 0x00000000;
+static const uint32_t g_qve_misc_select_mask = 0xFFFFFFFF;
+
+static const uint64_t g_qve_attribute_flags = 0x0000000000000001;
+static const uint64_t g_qve_attribute_xfrm = 0x0000000000000000;
+static const uint64_t g_qve_attribute_flags_mask = 0xFFFFFFFFFFFFFFFb;
+static const uint64_t g_qve_attribute_xfrm_mask = 0x0000000000000000;
+
+static const uint8_t g_qve_mrsigner[32] = {
+    0x8C, 0x4F, 0x57, 0x75, 0xD7, 0x96, 0x50, 0x3E, 0x96, 0x13, 0x7F,
+    0x77, 0xC6, 0x8A, 0x82, 0x9A, 0x00, 0x56, 0xAC, 0x8D, 0xED, 0x70,
+    0x14, 0x0B, 0x08, 0x1B, 0x09, 0x44, 0x90, 0xC5, 0x7B, 0xFF};
+
+static const uint16_t QVE_PRODID = 2;
+
+// Defense in depth, threshold must be greater or equal to least QvE ISV SVN
+const uint16_t LEAST_QVE_ISVSVN = 3;
+
+static void dump_info(
+    const char* title,
+    const uint8_t* data,
+    const uint8_t count)
+{
+    OE_TRACE_INFO("%s\n", title);
+    for (uint8_t i = 0; i < count; i++)
+    {
+        OE_TRACE_INFO("[%d] = %x\n", i, data[i]);
+    }
+}
+
+#if !defined(OE_USE_BUILTIN_EDL)
+/**
+ * Declare the prototype of the following function to avoid the
+ * missing-prototypes warning.
+ */
+
+oe_result_t _oe_verify_quote_ocall(
+    oe_result_t* _retval,
+    const oe_uuid_t* format_id,
+    const void* opt_params,
+    size_t opt_params_size,
```

Comment:
nit: Should this be "optional_parameters"?  https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md says:

> DON'T use abbreviations unless they are already well-known terms known by users (e.g., "app", "info"), or are already required for use by developers (e.g., "min", "max", "args"). Examples of bad use would be `num_widgets` instead of `widget_count`, and `opt_widgets` instead of `option_widgets` or `optional_widgets`.

Same question for line 93

Reply:
I agree with you.
But there are about 339 `opt_params` in OE project, I think it's not a PR specific issue. 
I suggest to update abbreviations in overall project with a single PR.